packages/react/src/UnderlineNav/styles.ts

```diff
@@ -135,17 +137,34 @@ export const menuItemStyles = {
   // To reset the style when the menu items are rendered as react router links
   textDecoration: 'none',
 }
+/**
+ *
+ * @param containerRef The Menu List Container Reference.
+ * @param listRef The Underline Nav Container Reference.
+ * @description This is the styles for our popover menu for the underline nav.
+ */
+export const menuStyles = (containerRef: Element | null, listRef: Element | null): BetterSystemStyleObject => {
+  const minStyleWidth = 192
 
-export const menuStyles = {
-  position: 'absolute',
-  zIndex: 1,
-  top: '90%',
-  right: '0',
-  boxShadow: '0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24)',
-  borderRadius: '12px',
-  backgroundColor: 'canvas.overlay',
-  listStyle: 'none',
-  // Values are from ActionMenu
-  minWidth: '192px',
-  maxWidth: '640px',
+  const baseRules = {
+    position: 'absolute',
+    zIndex: 1,
+    top: '90%',
+    boxShadow: '0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24)',
+    borderRadius: '12px',
+    backgroundColor: 'canvas.overlay',
+    listStyle: 'none',
+    // Values are from ActionMenu
+    minWidth: `${minStyleWidth}px`,
+    maxWidth: '640px',
+  }
+
+  // Do we have less room in the UnderlineNav container then what we have reserved with our minWidth?
+  // If so, do not position the element using right, as it will overflow. Let's use `getAnchoredPosition` instead.
+  if (containerRef && listRef && listRef.clientWidth < minStyleWidth) {
+    const {left} = getAnchoredPosition(containerRef, listRef, {align: 'center', side: 'outside-left'})
```

Comment:
I also played with these align and side values and updated the side to be `outside-bottom` because I think `outside-left` was causing an unexpected shift when the menu is open. (I am not 100% sure so let me know if I am missing anything here.) Here is a screen recording to explain the behaviour.

https://github.com/primer/react/assets/1446503/4097b7a7-ab5f-452c-b5b6-ab0dc9fd5a7d

_Video preview: example underlinenav menu has 'README', 'Code of Conduct', 'MIT Licence', 'Security' options with the two icon buttons aligned on the right side. The overflow menu appears when there is an overflow as the screen shrinks towards the left side. Overflow menu pops up towards the left side with its right edge aligning with the more button's right edge. When the screen gets small enough, the menu shifts towards the right side and its left edge now is aligning with the more button's right edge. As screen gets smaller, the menu repositions itself to bottom center._

```jsx
const {left} = getAnchoredPosition(containerRef, listRef, {align: 'start', side: 'outside-bottom'})
```

Also updating `align`  to `'start'` helps us make sure the menu is always aligning with the beginning of the list when there is not enough padding/margin around the navigation.

Reply:
Oh thank you thank you thank you! I was struggling with this, I couldnt find a combination that would work better then aligning it `center` and using `outside-bottom`. Your solution seem to have fixed what I was struggling with however, appreciate it :pray: